### PR TITLE
#266: refactor code after bumping fabric8 client version

### LINKED ISSUE #266: Support fabric8 5.0.0
[kind/Enhancement]

## Which functionality do you think we should update/improve?
update fabric8 dependency to 5.0.0

## Why is this needed? 
use of redhat-developer/intellij-common requires fabric8 5.0.0

### PR DESCRIPTION
**Pull request by sbouchet**
Signed-off-by: Stephane Bouchet <sbouchet@redhat.com>

## What is the purpose of this change? What does it change?
refactor code and bump fabric8 client to 5.0.0

## Was the change discussed in an issue?
fixes #266

### CODE REVIEW

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java [hunk 1]
```diff
@@ -220,14 +218,10 @@ public void logout() throws IOException {
...
-                        ComponentInfo.Builder builder = new ComponentInfo.Builder();
-                        ComponentInfo info = builder.withComponentKind(kind).build();
-                        components.add(Component.of(comp.getName(), ComponentState.NOT_PUSHED, path, info));
-                    } catch (IOException e) {
-                       throw e;
-                    }
+                    ComponentKind kind = getComponentKind(path);
+                    ComponentInfo.Builder builder = new ComponentInfo.Builder();
+                    ComponentInfo info = builder.withComponentKind(kind).build();
+                    components.add(Component.of(comp.getName(), ComponentState.NOT_PUSHED, path, info));
```
Comment (adietish) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java:224]: these are the only lines added, they're not catched any more. The rest are removals. Can you please elaborate how these removals fix things when using the newer kubernetes-client?

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java:224]: this is an unnecessary try-catch clause. the method already throws IOE and catch clause only re-throw IOE.

**Comment by adietish** [comment]
+1, works for me with the additional explanations about the code removals.

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**